src/testing/tree_construction/parser.rs

```diff
@@ -296,7 +297,7 @@ fn test(i: Span) -> IResult<Span, TestSpec> {
 
             TestSpec {
                 position,
-                data: data.to_string(),
+                data: data.to_string().trim_matches(|c| c == '\n').to_string(),
```

Comment:
There are some nom parse issues when there is a \n at the end of the data. I can't seem to get the parser fix this during parsing, so we do this as a post-step. It works, but I reckon this can be done better.

Reply:
Post-processing what the nom parser has a hard time with seems fine.  As long as `TestSpec` isn't losing any important information, the wrapper `Test` class can work it into a suitable format.